kanidm_client/tests/default_entries.rs

```diff
@@ -98,6 +98,7 @@ fn is_attr_writable(rsclient: &KanidmClient, id: &str, attr: &str) -> Option<boo
         ),
         entry => {
             let new_value = match entry {
+                "mail" => format!("{}@example.com", id),
```

Comment:
This is a test case where we are changing/adapting the value of the attribute based on well ... which attribute it is. Because we send the attribute to the server as a string, the str match makes sense. it *could* be an enum, but that enum would need to encapsulate all possible attributes that the server has in it's schema - and of course that schema adapts and grows over time. Which is where the problems start.

So actually, by being strings, we are "close to" the experience tthat the current version of the api has for users, and it also promotes fragility in the tests - we want to be sensitive and fail fast, which in this situation, is exactly what happened to lead me to creating a value for this.

Also the hardcoded example.com doesn't matter - it's a test case :)

Reply:
Tests should be fragile - they should fail fast when something - anything - changes. We don't want robust tests that self-heal, and fix things magically, or accidentally work when they shouldn't. We want them to be fragile and sensitive so they fail in slight changes so that we can understand and react to these in development.

Reply:
Ohh and I mean the JSON HTTP api version here with v1